File: 프로그래머스/0/181928. 이어 붙인 수/이어 붙인 수.js
function solution(num_list) {
    let oddStr = '';
    let evenStr = '';

    num_list.forEach(num => {
        if (num % 2 === 0) {
            evenStr += num.toString();
        } else {
            oddStr += num.toString();
        }
    });
    let oddNum = parseInt(oddStr, 10);
    let evenNum = parseInt(evenStr, 10);

    return oddNum + evenNum;
}

File: 프로그래머스/0/181928. 이어 붙인 수/이어 붙인 수.py
def solution(num_list):
    odd_str = ''
    even_str = ''

    for num in num_list:
        if num % 2 == 0:
            even_str += str(num)
        else:
            odd_str += str(num)

    odd_num = int(odd_str) if odd_str else 0
    even_num = int(even_str) if even_str else 0

    return odd_num + even_num